### PR TITLE
refactor bills by query method.

### PR DESCRIPTION
**Pull request by jcjurado3**
# high level what you did:
- The bills by query method wasn't caching the response

## Details:
Update from this method:
```
def bills_by_query
    bills_key = ["bills_by_query", @state_abbv, @query_keyword]
    response = Rails.cache.fetch(bills_key) do
      fetch_bills_by_query_from_api
    end
    json = JSON.parse(response.body, symbolize_names: true)
    bill_search_result = json[:searchresult]

    json
  end
```
to this method:
```
def bills_by_query
  bills_key = ["bills_by_query", @state_abbv, @query_keyword]
  response = Rails.cache.fetch(bills_key) do
    fetch_bills_by_query_from_api
  end

  if response&.body
    json = JSON.parse(response.body, symbolize_names: true)
    bill_search_result = json[:searchresult]
    json
  else
    # Handle the nil response appropriately
  end
end

```

## Overall Checklist:
- [ ] tests written for all methods added
- [ ] test coverage above 90%
- [ ] RSpec running clean
- [ ] ran rails s -- features working properly!
- [ ] pulled/merged milestone branch before submitting PR

### CODE REVIEW

File: app/services/bill_service.rb [hunk 1]
```diff
@@ -10,10 +10,14 @@ def bills_by_query
...
-    bill_search_result = json[:searchresult]
-
-    json
+  
+    if response&.body
+      json = JSON.parse(response.body, symbolize_names: true)
+      bill_search_result = json[:searchresult]
+      json
+    else
+      # Handle the nil response appropriately
```
Comment (aplasket) [inline app/services/bill_service.rb:19]: can we get rid of the else since we don't have anything in it?

Comment (aplasket) [reply app/services/bill_service.rb:19]: or are you still working through this?

**Review by aplasket — CHANGES_REQUESTED** [review]

File: app/services/bill_service.rb [hunk 2]
```diff
@@ -10,10 +10,14 @@ def bills_by_query
     response = Rails.cache.fetch(bills_key) do
       fetch_bills_by_query_from_api
     end
-    json = JSON.parse(response.body, symbolize_names: true)
-    bill_search_result = json[:searchresult]
-
-    json
+  
+    if response&.body
+      json = JSON.parse(response.body, symbolize_names: true)
```
Comment (aplasket) [inline app/services/bill_service.rb:15]: should this say `if response.body` -- does it need the & ?

**Review by aplasket — CHANGES_REQUESTED** [review]

File: app/services/bill_service.rb (hunk 2 above)
Comment (Jredish11) [reply app/services/bill_service.rb:15]: I think its some sort of ruby operator that checks if its nil. could be wrong

**Review by Jredish11 — APPROVED** [review]

**Review by aplasket — APPROVED** [review]

**Review by JCBeldo — APPROVED** [review]
Hope this fixes things!

**Pull request merged**